File: day01.py
left = []
right = []

with open("inputs/01.txt") as file:
    for line in file.readlines():
        l, r = map(int, line.split("   "))
        left.append(l)
        right.append(r)

left = sorted(left)
right = sorted(right)

distance = similarity = 0

for index, item in enumerate(left):
    distance += abs(item - right[index])
    similarity += right.count(item) * item

with open("outputs/01.txt", "w") as file:
    file.write(f"Part 1: {distance}\nPart 2: {similarity}")

File: src/bin/day01.rs
use std::fs;

pub fn main() {
    let input = fs::read_to_string("inputs/01.txt").unwrap();
    let mut left: Vec<i32> = vec![];
    let mut right: Vec<i32> = vec![];

    for line in input.lines() {
        let parts: Vec<i32> = line.split("   ").map(|part| {part.parse::<i32>().unwrap()}).collect();
        left.push(parts[0]);
        right.push(parts[1]);
    }

    left.sort();
    right.sort();

    let mut distance = 0;
    let mut similarity = 0;

    for (index, item) in left.iter().enumerate() {
        distance = distance + (item - right[index]).abs();
        similarity = similarity + (
            *item as usize * right.iter().filter(|&n| n == &left[index]).count()
        );
    }

    fs::write(
        "outputs/01.txt",
        format!("Part 1: {}\nPart 2: {}", distance, similarity)
    ).unwrap();
}
